src/components/lightDarkTheme/hook.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+import { dark, light } from 'src/constants';
+
+
+export const useLightDarkTheme = () => {
+
+    const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
+    const [theme, setTheme] = useState(localStorage.getItem('theme') || defaultTheme);
```

Comment:
could be case, when user opens App for first time and their browser doesn't support `window.matchMedia('(prefers-color-scheme: dark)')`. In this case theme === undefined. Better to set there third optional case:
``` JS
      const [theme, setTheme] = useState(localStorage.getItem('theme') || defaultTheme || THEMES.light);
```